weave/wandb_interface/wandb_stream_table.py

```diff
@@ -0,0 +1,249 @@
+import contextlib
+import json
+import logging
+import os
+import random
+import shutil
+import typing
+import uuid
+
+from wandb.sdk.lib.paths import LogicalPath
+
+from .wandb_lite_run import InMemoryLazyLiteRun
+
+from .. import runfiles_wandb
+from .. import storage
+from .. import weave_types
+from .. import artifact_base
+from .. import file_util
+
+if typing.TYPE_CHECKING:
+    from wandb.sdk.internal.file_pusher import FilePusher
+
+# Shawn recommended we only encode leafs, but in my testing, nested structures
+# are not handled as well in in gorilla and we can do better using just weave.
+# Uncomment the below to use gorilla for nested structures.
+TRUST_GORILLA_FOR_NESTED_STRUCTURES = False
+
+# Weave types are parametrized, but gorilla expects just simple strings. We could
+# send the top-level string over the wire, but this fails to encode type specifics
+# and therefore loses information. With this flag, we instead stringify the JSON type
+# and send that over the wire. This is a bit of a hack, but it works.
```

Comment:
Works but it will cause the data to be kinda huge

Reply:
Well in either case we need to store the type details - but you are correct that the type string might be huge.

It would be much nicer if gorilla stripped off the `_type` property from each cell!